spec/wizards/placements/add_placement/steps/year_group_spec.rb

```diff
@@ -0,0 +1,38 @@
+require "rails_helper"
+
+RSpec.describe Placements::AddPlacement::Steps::YearGroup, type: :model do
+  describe "attributes" do
+    it { is_expected.to have_attributes(school: nil, year_group: nil) }
+  end
+
+  describe "delegations" do
+    it { is_expected.to delegate_method(:primary?).to(:school).with_prefix }
+  end
+
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:school) }
+
+    context "when the school is primary" do
+      subject { described_class.new(school:) }
+
+      let(:school) { create(:placements_school, :primary) }
```

Comment:
Micro-optimisation: it may be possible to use `build` here rather than `create`, to avoid needing to write to the database.